src/main/java/org/broadinstitute/hellbender/tools/walkers/variantutils/SelectVariants.java

```diff
@@ -617,11 +632,28 @@ public void apply(VariantContext vc, ReadsContext readsContext, ReferenceContext
                     (!selectRandomFraction || Utils.getRandomGenerator().nextDouble() < fractionRandom)) {
                 //remove annotations being dropped and write variantcontext
                 final VariantContext variantContextToWrite = buildVariantContextWithDroppedAnnotationsRemoved(filteredGenotypeToNocall);
-                vcfWriter.add(variantContextToWrite);
+                if (variantContextToWrite.getStart() != vc.getStart()) {
```

Comment:
I would just add every variant to `pendingVariants`.  This check is redundant with the while loop at the beginning of `apply()`.  It's true that written as is the priority queue is usually empty, but I don't think the optimization is worth the added complexity.

Reply:
Yeah you're right, I was trying to avoid doing too many comparisons in the priority queue, but since it is nearly always empty this is an essentially useless optimization.